fix: Fix GraphQL query arguments for getPledgesByUserId query

ashutoshsao opened the pull request:
<!--
This section can be deleted after reading.

We employ the following branching strategy to simplify the development process and to ensure that only stable code is pushed to the `master` branch:

- `develop`: For unstable code: New features and bug fixes.
- `master`: Where the stable production ready code lies. Only security related bugs.

NOTE!!!

ONLY SUBMIT PRS AGAINST OUR `DEVELOP` BRANCH. THE DEFAULT IS `MAIN`, SO YOU WILL HAVE TO MODIFY THIS BEFORE SUBMITTING YOUR PR FOR REVIEW. PRS MADE AGAINST `MAIN` WILL BE CLOSED.

-->

<!--
Thanks for submitting a pull request! Please provide enough information so that others can review your pull request.
-->

**What kind of change does this PR introduce?**
Bugfix
<!-- E.g. a bugfix, feature, refactoring, etc… -->

**Issue Number:**
Fixes #5001

**Snapshots/Videos:**
<img width="1893" height="946" alt="screenshot-2025-12-14_15-06-58" src="https://github.com/user-attachments/assets/44415a70-b540-49a2-8a48-6f78a797e33e" />

<!--Add snapshots or videos wherever possible.-->

**If relevant, did you update the documentation?**
N/A
<!--Add link to Talawa-Docs.-->

**Summary**

Fixes GraphQL query argument mismatches for the "My Pledges" page and ensures i18n compliance by replacing all hardcoded strings with translation keys (addressing PR #4956 requirements).

**Root Cause:**
- `getPledgesByUserId` query used incorrect argument structure (expected `input` parameter, was passing `userId` directly)
- Query attempted to fetch non-existent `users` field from `FundCampaignPledge` type
- `USER_DETAILS` query used incorrect variable structure and accessed nested response that no longer exists

**Changes Made:**

1. **Fixed USER_PLEDGES query** (`src/GraphQl/Queries/fundQueries.ts`):
   - Changed to use `input: $input` instead of `userId: $userId`
   - Removed non-existent `users` field

2. **Fixed PledgeModal** (`src/screens/UserPortal/Campaigns/PledgeModal.tsx`):
   - Updated `USER_DETAILS` variables to `input: { id: 'userId' }` format
   - Fixed response access structure
   - Added i18n: Updated "Pledgers" label to use `t('pledgers')`

3. **Updated Pledges component** (`src/screens/UserPortal/Pledges/Pledges.tsx`):
   - Changed query variables to `input: { userId: userId }` format
   - Added i18n: Replaced hardcoded strings with translation keys for all table headers and labels

4. **Updated tests**:
   - Updated mocks to use `input` parameter format
   - Fixed tests for i18n compliance

5. **Added translation keys** (`public/locales/en/translation.json`):
   - Added keys: `pledger`, `associatedCampaign`, `pledged`, `donated`, `progress`, `more`, `pledgers`

6. **Rebased onto latest develop**:
   - Includes PR #4956 (i18n enforcement) requirements

<!-- Explain the **motivation** for making this change. What existing problem does the pull request solve? -->
<!-- Try to link to an open issue for more information. -->

**Does this PR introduce a breaking change?**
No
<!-- If this PR introduces a breaking change, please describe the impact and a migration path for existing applications. -->

## Checklist

### CodeRabbit AI Review
- [x] I have reviewed and addressed all critical issues flagged by CodeRabbit AI
- [x] I have implemented or provided justification for each non-critical suggestion
- [x] I have documented my reasoning in the PR comments where CodeRabbit AI suggestions were not implemented

### Test Coverage
- [x] I have written tests for all new changes/features
- [x] I have verified that test coverage meets or exceeds 95%
- [x] I have run the test suite locally and all tests pass


**Other information**
N/A
**Have you read the [contributing guide](https://github.com/PalisadoesFoundation/talawa-admin/blob/master/CONTRIBUTING.md)?**
Yes
<!--Yes or No-->

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

* **Refactor**
  * Standardized pledge/query input shapes, flattened pledger user data, removed multi-pledger popup, simplified pledge listing, and added safer navigation and loading guards.

* **Tests**
  * Updated mocks to the new input/response shape, removed legacy multi-pledger popup tests, and made autocomplete tests resilient across component versions and absent-option scenarios.

* **Localization / UI**
  * Added pledge-related translations across locales, switched labels to i18n, and adjusted UI composition and interaction behavior.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

palisadoes commented:
Please fix the failing test

ashutoshsao commented:
yes i am doing that

ashutoshsao commented:
done !

ashutoshsao commented:
@palisadoes done !

ashutoshsao commented:
done !

palisadoes reviewed (changes requested):
Please address this:
- https://github.com/PalisadoesFoundation/talawa-admin/pull/5024#pullrequestreview-3581391476

ashutoshsao commented:
hey @palisadoes i have addressed the review , i had a question for you, i asked to work on issue https://github.com/PalisadoesFoundation/talawa-admin/issues/4843 , the other guy, ignored that and started working it it isn't fair right ?

palisadoes commented:
@ashutoshsao You may now use the repo's issue auto-assignment feature outlined here: ISSUE_GUIDELINES.md

ashutoshsao commented:
> hey @palisadoes i have addressed the review , i had a question for you, i asked to work on issue first #4843 , the other guy ignored that and started working it, it isn't fair right ?

thank you, but i just wanted to know, if i was the first guy to ask to work on it shouldn't i be the one first addressed ?
